### PR TITLE
OCPBUGS-34807: Fixed ValidReleaseInfo condition

### PR DESCRIPTION
**Pull request by muraee**
**What this PR does / why we need it**:
Previously the condition was reconciled immediately before accessing any image in the playload. which means the condition will always be `True`

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go [hunk 1]
```diff
@@ -301,6 +301,19 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 	}
 
 	originalHostedControlPlane := hostedControlPlane.DeepCopy()
+	// Always update status based on the current state of the world.
+	defer func() {
```
Comment (enxebre) [inline control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:305]: I'm not a huge fan of defer and patching, how about we keep the existing structure and have these missing images to be called and patched explicitly in e.g. l768?

**Comment by enxebre** [comment]
This is reporting kuryr-cni and kuryr-controller from the cluster-network-operator deployment, @kyrtapz can we drop those references now?

**Comment by kyrtapz** [comment]
> This is reporting kuryr-cni and kuryr-controller from the cluster-network-operator deployment, @kyrtapz can we drop those references now?

Yes, kuryr is no longer a supported CNI.

File: control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go (hunk 1 above)
Comment (muraee) [reply control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go]: updated

**Comment by muraee** [comment]
requires https://github.com/openshift/hypershift/pull/4164

**Review by stevekuznetsov — APPROVED** [review]

**Pull request merged**